### DISCUSSION TITLE
Does pyright speed up if you give it more cores?

### DISCUSSION THREAD

**Discussion by dstromberg** [microsoft/pyright#4493] [General]
Hi folks.

I'm working on a ~62,000 line CPython project.

I recently decided to make the switch from vim+syntastic to lunarvim+pyright.

But I'm finding pyright a bit sluggish.  Perhaps other language servers would be even slower on a project of this size, I don't know.

The (virtual) machine I'm using it on has 4 CPU cores at the moment.  I could conceivably add more - if it'll help.

Does pyright take advantage of additional CPU cores?

Thanks for the cool tool.

**Comment by erictraut** [comment]
No, additional cores won't help. Type analysis is single threaded by design.

Adding type annotations will generally help improve performance. Type inference, especially for function return types, can be very expensive.
